src/libraries/System.Data.Common/src/System/Data/Common/DataStorage.cs

```diff
@@ -560,9 +560,9 @@ protected void SetNullStorage(BitArray nullbits)
         internal static Type GetType(string value)
         {
             Type? dataType = Type.GetType(value); // throwOnError=false, ignoreCase=fase
-            if (null == dataType)
+            if (dataType == null)
             {
-                if ("System.Numerics.BigInteger" == value)
+                if (value == "System.Numerics.BigInteger")
```

Comment:
`value == typeof(System.Numerics.BigInteger).FullName`?  Don't know all the implications of that though.

Reply:
There is a comment above the function:
```
/// Types like "System.Guid" will load regardless of AssemblyQualifiedName because they are special
/// Types like "System.Data.SqlTypes.SqlString" will load because they are in the same assembly as this code
/// Types like "System.Numerics.BigInteger" won't load because they are not special and not same assembly as this code
```

Reply:
I don't think that means we have to use the string, but I am not 100% sure.

Reply:
`typeof` has to work, because it already does it the line below anyways.

Reply:
Please let's limit this change to just reordering equality operators. Future changes are always possible, but making the PR(s) as simple and mechanical as possible makes it easier to review and less likely to introduce a bug.

Reply:
@danmosemsft are you okay if we create a separate PR for this? Or will we need an issue first?

Reply:
> @danmosemsft are you okay if we create a separate PR for this? Or will we need an issue first?

It's a judgement call, if something is a small change, and you're pretty sure it's a good one, sometimes people throw up a PR without an issue. We don't have a hard rule like some communities. But an issue is generally a good idea, as it avoids wasting time making a code change we wouldn't take or that would need reworking to a new approach.